src/states/node.rs

```diff
@@ -77,6 +74,21 @@ const CLIENT_BAN_DURATION: Duration = Duration::from_secs(2 * 60 * 60);
 /// Duration for which clients' IDs we disconnected from are retained.
 const DROPPED_CLIENT_TIMEOUT: Duration = Duration::from_secs(2 * 60 * 60);
 
+pub struct NodeDetails {
+    pub ack_mgr: AckManager,
+    pub cache: Box<Cache>,
+    pub chain: Chain,
+    pub crust_service: Service,
+    pub full_id: FullId,
+    pub gen_pfx_info: GenesisPfxInfo,
+    pub msg_queue: VecDeque<RoutingMessage>,
+    pub notified_nodes: BTreeSet<PublicId>,
+    pub parsec_map: ParsecMap,
+    pub peer_mgr: PeerManager,
+    pub routing_msg_filter: RoutingMessageFilter,
+    pub timer: Timer,
+}
+
 pub struct Node {
     ack_mgr: AckManager,
```

Comment:
Wouldn't it make sense to just use composition to avoid repeating all these fields?